### ISSUE TITLE
len_without_is_empty false positive

### ISSUE THREAD

**Issue by clarcharr** [rust-lang-nursery/rust-clippy#1740]
```
pub trait Empty {
    fn is_empty(&self) -> bool;
}
pub trait Len: Empty {
    fn len(&self) -> usize;
}
```

This triggers the lint. Implemented for [`len-trait`](https://docs.charr.xyz/len-trait/len_trait/len/trait.Len.html) crate though, so, it might be too niche of a case to cover.

**Comment by oli-obk** [comment]
Basically when we trigger the lint we also need to check traits required by the current trait for the existance of an `is_empty`. Should be doable.
